src/app/components/modal/modal.component.ts

```diff
@@ -57,17 +58,18 @@ export class ModalComponent extends BaseComponent implements OnInit {
             this.modalService
                 .getCommand()
                 .subscribe((command: ModalCommandInterface) => {
-                    if (command.context) {
-                        // @ts-ignore
+                    if (command?.context) {
                         command.context.buttonCount = 0;
                         [
                             "yesButtonText",
                             "noButtonText",
                             "lastButtonText",
                             "tooLastButtonText",
                         ].forEach((btnName: string) => {
+                            // eslint-disable-next-line @typescript-eslint/ban-ts-comment
                             // @ts-ignore
                             if (command.context[btnName]) {
+                                // eslint-disable-next-line @typescript-eslint/ban-ts-comment
                                 // @ts-ignore
                                 command.context.buttonCount += 1;
                             }
```

Comment:
Remove this entirely. Instead, compute number of present buttons as a function in the `ModalDialogComponent`:

```
    private getContextButtonsCount() : number {
        let buttonsCount = 0;
        if (this.context.yesButtonText) buttonsCount++;
        if (this.context.noButtonText) buttonsCount++;
        if (this.context.lastButtonText) buttonsCount++;
        if (this.context.tooLastButtonText) buttonsCount++;
        return buttonsCount;
    }
```

**And KILL the `buttonCount` property!**

Reply:
Removed the buttonCount property

Reply:
It was still present in `ModalCommandInterface`